### PR TITLE
Myriad/registry tests

### PR DESCRIPTION
**Pull request by nael-binary**

### CODE REVIEW

File: lib/Myriad/Registry.pm [hunk 1]
```diff
@@ -204,14 +175,17 @@ method add_emitter ($pkg, $method, $code, $args) {
     };
 }
 
-=head2 emitters_for
+=head2 emitter_for
```
Comment (ea-binary) [inline lib/Myriad/Registry.pm:178]: why did you change the name? it was more accurate

File: lib/Myriad/Registry.pm [hunk 2]
```diff
@@ -229,14 +203,17 @@ method add_receiver ($pkg, $method, $code, $args) {
     };
 }
 
-=head2 receivers_for
+=head2 receiver_for
```
Comment (ea-binary) [inline lib/Myriad/Registry.pm:206]: ditto

File: lib/Myriad/Registry.pm [hunk 3]
```diff
@@ -125,69 +129,36 @@ Returns a hashref of RPC definitions for the given class.
...
-
-=head2 sinks_for
-
-Returns a hashref of sink methods for the given class.
-
-=cut
-
-method sinks_for ($pkg) {
-    return $sink->{$pkg};
+method batch_for ($pkg) {
```
Comment (ea-binary) [inline lib/Myriad/Registry.pm:157]: ditto

File: lib/Myriad/Service/Implementation.pm [hunk 4]
```diff
@@ -172,7 +172,7 @@ async method start {
             return;
         }
 
-        if(my $emitters = $registry->emitters_for(ref($self))) {
+        if(my $emitters = $registry->emitter_for(ref($self))) {
```
Comment (ea-binary) [inline lib/Myriad/Service/Implementation.pm:175]: The changes here are not needed if you revert your changes earlier

File: t/config.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+multi_option:
```
Comment (ea-binary) [inline t/config.yml:1]: not very important, but it would be neat of you mock the `load_files` sub instead of adding this file

File: lib/Myriad/Registry.pm (hunk 1 above)
Comment (nael-binary) [reply lib/Myriad/Registry.pm:178]: I wanted to discuss that actually. I did it mainly to avoid irregular plurals. like `batches_for`
one thing also, for RPC we have it as `rpc_for` not `rpcs_for`

I do agree that `emitters_for` sounds way better to what it does as it returns a list of them.
However I do believe that their names should have some consistency. Maybe add `all` somewhere to the name?

File: t/config.yml (hunk 5 above)
Comment (nael-binary) [reply t/config.yml:1]: hmm I think its ok for reading file. the Multi option here is purely to test this functionality added in code even though we don't use it for anything yet.

File: lib/Myriad/Registry.pm (hunk 1 above)
Comment (ea-binary) [reply lib/Myriad/Registry.pm:178]: yes, `rpc` is a special case no need to make everything a special case :sweat_smile:

File: t/config.yml (hunk 5 above)
Comment (ea-binary) [reply t/config.yml:1]: understood, but you'd find people argue that you should isolate your tests even from filesystem access, as I said not very important.

File: lib/Myriad/Registry.pm (hunk 1 above)
Comment (nael-binary) [reply lib/Myriad/Registry.pm:178]: `batch(es)` too

File: lib/Myriad/Registry.pm [hunk 6]
```diff
@@ -31,13 +31,19 @@ declare_exception UnknownClass => (
...
-has $emitter = {};
-has $receiver = {};
+has $rpc ;
+has $service_by_name;
+has $batch;
+has $emitter;
+has $receiver;
+
+BUILD {
+ $rpc = {};
```
Comment (leonerd) [inline lib/Myriad/Registry.pm:41]: Any particular reason why you're storing these as hashrefs? You could just

```perl
has %rpc;
```

and save yourself the trouble in the `BUILD` block
